lib/scholar/model_selection.ex

```diff
@@ -105,9 +106,129 @@ defmodule Scholar.ModelSelection do
         ]
       >
   """
-  def cross_validate(x, y, folding_fun, scoring_fun) do
+  def cross_validate(x, y, folding_fun, scoring_fun, opts \\ []) do
```

Comment:
Do we want to support version without options?

```elixir
cond do
  is_function(scoring_fun, 3) ->
    fn {x, y} -> scoring_fun.(x, y, opts) |> Nx.stack() end
  is_function(scoring_fun, 2) -> 
    if opts != [] do
      raise ArgumentError, "no options must be given if scoring_fun expects two arguments"
    end
    fn {x, y} -> scoring_fun.(x, y) |> Nx.stack() end
  true ->
    raise ArgumentError, "expected scoring_fun to be a function of arity 2 or 3, got: #{inspect(scoring_fun)}"
end
```

Reply:
Wait a second... we don't process the opts in anyway. So I don't see why we need to pass them here? If you need to pass options to the scoring_fun, you can just close over it:

```
     iex> scoring_fun = fn x, y ->
      ...>   {x_train, x_test} = x
      ...>   {y_train, y_test} = y
      ...>   model = Scholar.Linear.LinearRegression.fit(x_train, y_train, some_option: 123)
```

I think we can remove the `opts` argument in the cross validation functions now that we have weighted cross validate.

Reply:
I also posted an example on grid search. :)